lib/internal/Magento/Framework/Reflection/DataObjectProcessor.php

```diff
@@ -128,7 +136,7 @@ public function buildOutputDataArray($dataObject, $dataObjectType)
                 }
             }
 
-            $outputData[$key] = $value;
+            $outputData[$this->mapObjectKey($key, ltrim($dataObjectType, '\\'))] = $value;
```

Comment:
Can we move ltrim inside the mapObjectKey method?

Reply:
We could. But that wouldn't make any difference to the flow right? The mapObjectKey should only be responsible for mapping the object keys. Which is why I had kept the ltrim (to include any class names with preceding slashes given in the di.xml as well) outside of it

Reply:
The issue with it that it might not find the correct key, if in the client code we'll not do ltrim.

Reply:
I don't think I understand. The `mapObjectKey` can't be called from any other class. It'll just be called from the `buildOutputDataArray` function, and we have done the `ltrim` there

Reply:
Ok. but in case if someone will adjust this code a bit later or will call it in the another method - this issue will appear. It's better to incapsulate this logic inside the method

Reply:
I see. Okay. In that case I think it's better that we name the function something other than `mapObjectKey`?

Reply:
getKeyByObjectType?

Reply:
Guess that'll work